app/uk/gov/hmrc/economiccrimelevystubs/services/EclRegistrationReferenceService.scala

```diff
@@ -34,4 +35,14 @@ class EclRegistrationReferenceService @Inject() (sequenceRepository: SequenceRep
       s"$prefix${"%010d".format(id)}"
     }
 
+  def getPreviousReferences: Future[List[String]] =
+    sequenceRepository.getCurrentReference(eclRegistrationReferenceKey).map {
+      case Some(ref) =>
+        val referenceRange = List.range(1, ref + 1)
+        val references     = new ListBuffer[String]()
+        for (elem <- referenceRange)
+          references += s"$prefix${"%010d".format(elem)}"
```

Comment:
Can I suggest putting `"%010d"` in a private val in the class as it's also used elsewhere

Reply:
Done.